tests/json-fixtures/test_transactions.py

```diff
@@ -40,68 +45,90 @@
-    vm = chain.get_vm()
-    TransactionClass = vm.get_transaction_class()
+@pytest.fixture
+def fixture_transaction_class(fixture_data):
+    _, _, fork_name = fixture_data
+    if fork_name == ForkName.Frontier:
+        return FrontierTransaction
+    elif fork_name == ForkName.Homestead:
+        return HomesteadTransaction
+    elif fork_name == ForkName.EIP150:
+        return HomesteadTransaction
+    elif fork_name == ForkName.EIP158:
+        return SpuriousDragonTransaction
+    elif fork_name == ForkName.Byzantium:
+        return ByzantiumTransaction
+    elif fork_name == ForkName.Sharding:
+        pytest.skip("Sharding Transaction class has not been implemented")
+    elif fork_name == ForkName.Constantinople:
+        pytest.skip("Constantinople Transaction class has not been implemented")
+    elif fork_name == ForkName.Metropolis:
+        pytest.skip("Metropolis Transaction class has not been implemented")
+    else:
+        raise ValueError("Unknown Fork Name: {0}".format(fork_name))
 
-    if 'sender' in fixture:
-        transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        expected = normalize_signed_transaction(fixture['transaction'])
-
-        assert transaction.nonce == expected['nonce']
-        assert transaction.gas_price == expected['gasPrice']
-        assert transaction.gas == expected['gasLimit']
-        assert transaction.to == expected['to']
-        assert transaction.value == expected['value']
-        assert transaction.data == expected['data']
-        assert transaction.v == expected['v']
-        assert transaction.r == expected['r']
-        assert transaction.s == expected['s']
 
-        sender = to_canonical_address(fixture['sender'])
+def test_transaction_fixtures(fixture, fixture_transaction_class):
 
-        try:
-            assert transaction.sender == sender
-        except BadSignature:
-            assert not (27 <= transaction.v <= 34)
-    else:
-        # check RLP correctness
-        try:
-            transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        # fixture normalization changes the fixture key from rlp to rlpHex
-        except KeyError:
-            assert fixture['rlpHex']
-            return
-        # rlp is a list of bytes when it shouldn't be
-        except TypeError as err:
-            assert err.args == ("'bytes' object cannot be interpreted as an integer",)
-            return
-        # rlp is invalid or not in the correct form
-        except (rlp.exceptions.ObjectDeserializationError, rlp.exceptions.DecodingError):
-            return
+    TransactionClass = fixture_transaction_class
 
+    try:
+        txn = rlp.decode(fixture['rlp'], sedes=TransactionClass)
+    except (rlp.DeserializationError, rlp.exceptions.DecodingError):
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    except TypeError as err:
+        assert err.args == ("'bytes' object cannot be interpreted as an integer",)
```

Comment:
Can you add a comment which indicates what this is caused by?

Reply:
Yep! Updated